lib/travis/build/script/r.rb

```diff
@@ -0,0 +1,368 @@
+            sh.cmd 'sudo installer -pkg "/tmp/R-latest.pkg" -target /'
+            sh.rm '/tmp/R-latest.pkg'
+
+          else
+            sh.failure "Operating system not supported: #{config[:os]}"
+          end
+
+          if needs_bioc?
+            setup_bioc
+          end
+          if config[:latex]
+            setup_latex
+          end
+          if config[:pandoc]
+            setup_pandoc
+          end
+        end
+
+        def announce
+          super
+
+          sh.cmd 'Rscript -e \'sessionInfo()\''
+          sh.echo ''
+        end
+
+        def install
+          super
+
+          # Install any declared packages
+          apt_install config[:apt_packages]
+          r_binary_install config[:r_binary_packages]
+          r_install config[:r_packages]
+          bioc_install config[:bioc_packages]
+          r_github_install config[:r_github_packages]
+
+          # Install dependencies for the package we're testing.
+          install_deps
+        end
+
+        def script
+          # Build the package
+          sh.echo "Building with: R CMD build ${R_BUILD_ARGS}"
+          sh.cmd "R CMD build #{config[:r_build_args]} ."
+          tarball_script = [
+            'library("devtools")',
+            'pkg <- as.package(".")',
+            'cat(paste0(pkg$package, "_", pkg$version, ".tar.gz"))',
+          ].join('; ')
+          sh.export 'PKG_TARBALL', "$(Rscript -e '#{tarball_script}')"
+
+          # Test the package
+          sh.echo 'Testing with: R CMD check "${PKG_TARBALL}" ' +
+                  "#{config[:r_check_args]}"
+          sh.export '_R_CHECK_CRAN_INCOMING_', as_r_boolean(
+                      config[:r_check_cran_incoming])
+          if config[:r_check_cran_incoming]
+            sh.echo "(CRAN incoming checks are off)"
+          end
+          sh.cmd "R CMD check \"${PKG_TARBALL}\" #{config[:r_check_args]}"
+
+          # Turn warnings into errors, if requested.
+          if config[:warnings_are_errors]
+            export_rcheck_dir
+            sh.if 'grep -q -R "${RCHECK_DIR}/**/*.00check" "WARNING"' do
+              sh.failure "Found warnings, treating as errors (as requested)."
+            end
+          end
+          
+          # Check revdeps, if requested.
+          if config[:r_check_revdep]
+            sh.echo "Checking reverse dependencies"
+            revdep_script = [
+              'library("devtools");',
+              'res <- revdep_check();',
+              'if (length(res) > 0) {',
+              ' revdep_check_summary(res);',
+              ' revdep_check_save_logs(res);',
+              ' q(status = 1, save = "no");',
+              '}',
+            ].join(' ')
+            sh.cmd "Rscript -e '#{revdep_script}'"
+          end
+
+        end
+
+        def after_failure
+          dump_logs
+          super
+        end
+
+        private
+
+        def needs_bioc?
+          config[:bioc_required] or not config[:bioc_packages].empty?
```

Comment:
hah! i had no idea they were different in ruby. (in fact, i didn't even know there was a `||` operator -- i typed `or` and emacs highlighted it, so i figured that was the same as python.)

i think i'm safe here, but having read that article, i switched to `||` anyway.